### DISCUSSION TITLE
ELN Consortium Meeting 4

### DISCUSSION THREAD

**Discussion by NicolasCARPi** [TheELNConsortium/TheELNFileFormat#55] [General]
# When?

March 11th 2024 at 4PM CET

# What?

* discussion about cryptographic signature of the .eln to prove authenticity
* other topics to come, please comment!

Previous meeting notes: https://github.com/TheELNConsortium/TheELNFileFormat/discussions/39#discussioncomment-7836409

**Comment by NicolasCARPi** [comment]
It's today at 4PM CET!

https://zoom.us/j/95736282457?pwd=eE51QlFPZklqSUpxQk9MU2szdGlNUT09

**Comment by NicolasCARPi** [comment]
# minutes

## How to sign a .eln with cryptographic signature so that we can trust it during import

issue: https://github.com/TheELNConsortium/TheELNFileFormat/issues/56
Florian will try something with openssl commands.

## Additional metadata

Where to store metadata that could not be added/expressed in json-ld metadata file?
Ideally it is not needed anymore and we can have everything expressed in the json-ld file.
Discussion here: https://github.com/TheELNConsortium/TheELNFileFormat/issues/58
Nicolas C. will try and have some examples with these property values thing to express extra fields.

## Multiple authors

When a Dataset has multiple authors, how to express the idea that one is the actual main author and another is the creator of the crate, and maybe a third is a contributor. See https://schema.org/creator

## How to discriminate experiments, samples, equipements

Use `genre`: https://schema.org/genre

SPECIFICATION will need to be updated to list the accepted values:

- experiment
- resource
- sample
- measurement
- simulation
- procedure
- instrument

If no `genre` is present, user can select it before import, or the software has a default genre.

## History of Dataset/File

Use https://schema.org/UpdateAction and https://schema.org/CreateAction to convey history information.

## How to check if intent has been kept

Import, then export and verify that basically only timestamps have been modified. If the exported json-ld has different properties than source file, it means some things were misinterpreted.

**Reply by FlorianRhiem** [reply]
Thank you for the minutes!

Small correction, I will not try to use zipsign, but instead use the OpenSSL CMS implementation similar to zipsign.

Also, I think we agreed to use `creator` instead of `accountablePerson`, as legal accountability often is handled differently.

Is there already an example of UpdateAction and CreateAction?

**Reply by NicolasCARPi** [reply]
This is what can be found now in elab eln:

![2024-03-12-095113_559x313_scrot](https://github.com/TheELNConsortium/TheELNFileFormat/assets/3043706/f028c048-8f68-41d3-9c9b-b3c034c6e6c5)

FYI, we have this so we can discriminate the producing version and have different strategy during import based on that.
